src/jdk.jdwp.agent/share/native/libjdwp/commonRef.c

```diff
@@ -200,16 +219,10 @@ weakenNode(JNIEnv *env, RefNode *node)
 
         if (weakRef != NULL) {
             JNI_FUNC_PTR(env,DeleteGlobalRef)(env, node->ref);
-            node->ref         = weakRef;
-            node->strongCount = 0;
-        }
-        return weakRef;
```

Comment:
In this case I don't see that the removal of this return is correct if `weakRef==NULL` ?

Reply:
Similar issue here. It should be returning NULL in this case, not `node->ref.`